server/schedulers/balance_leader.go

```diff
@@ -192,6 +296,8 @@ func (cs *candidateStores) reSort(stores ...*core.StoreInfo) {
 }
 
 func (l *balanceLeaderScheduler) Schedule(cluster schedule.Cluster) []*operator.Operator {
+	l.conf.mu.RLock()
```

Comment:
How about locking it only when it's going to be read?

Reply:
I add a var `batch` to read it. After reading,  unlock it immediately